lib/resty/etcd/v3.lua

```diff
@@ -160,6 +194,37 @@ local function choose_endpoint(self)
     return endpoints[pos]
 end
 
+-- return refresh_is_ok, error
+function refresh_jwt_token(self)
+    -- token exist and not expire
+    -- default is 5min, we use 3min
+    -- https://github.com/etcd-io/etcd/issues/8287
+    if self.jwt_token and now() - self.last_auth_time < 60 * 3 then
+        return true, nil
+    end
+
+    local opts = {
+        body = {
+            name         = self.user,
+            password     = self.password,
+        }
+    }
+    local res, err = _request_uri(self, 'POST',
+                        choose_endpoint(self).full_prefix .. "/auth/authenticate",
+                        opts, 5, true)    -- default authenticate timeout 5 second
+    if err then
+        return nil, err
+    end
+
+    if not res or not res.body or not res.body.token then
+        return nil, 'authenticate refresh token fail'
+    end
+
+    self.jwt_token = res.body.token
```

Comment:
When we access an etcd cluster, can this JWT_TOKEN be used on different etcd nodes?

Reply:
yes the etcd cluster will sync the auth